src/khoj/processor/conversation/gpt4all/chat_model.py

```diff
@@ -0,0 +1,137 @@
+from typing import Union, List
+from datetime import datetime
+import sys
+import logging
+from threading import Thread
+
+from langchain.schema import ChatMessage
+
+from gpt4all import GPT4All
+
+
+from khoj.processor.conversation.utils import ThreadedGenerator, generate_chatml_messages_with_context
+from khoj.processor.conversation import prompts
+from khoj.utils.constants import empty_escape_sequences
+
+logger = logging.getLogger(__name__)
+
+
+def extract_questions_falcon(
```

Comment:
Is Falcon able to extract questions from the user's message? And in valid json format? It maybe hard as GPT-3.5 has trouble with passing back well formated json as well (which is one of the things OpenAI's [function calling](https://platform.openai.com/docs/guides/gpt/function-calling) was created to fix)

Reply:
It's not able to do so. I've left the method in because I think the interface of our different models should implement these functions, but by default it just does some minimal string manipulation to extract the literal questions from the user's query. For example, `"Where did I go to school? Which team did I play on?"` would be separated into `["Where did I go to school?", "Which team did I play on?"]`.